Additional fixes to the docker documentation (#65)
StrangeBeeCorp/docs: docs/thehive/installation/docker/docker-compose.md

---

## Prerequisites

Ensure the following are installed on your Ubuntu system:

- **Docker**: A platform to develop, ship, and run applications in containers.
- **Docker Compose**: A tool for defining and running multi-container Docker applications.

---

## Setting Up TheHive with Docker Compose

1. **Create a dedicated directory for TheHive setup:**

    ```bash
    mkdir -p ~/thehive
    cd ~/thehive
    ```

2. **Create a configuration directory for TheHive:**

    Create a directory inside ``~/thehive`` folder:

    ```bash
    mkdir -p ~/thehive/thehive/config
    ```

    This directory will be used to store TheHive configuration files, ensuring they are properly organized and easily accessible.

4. **Create the configuration files:**

    **`application.conf`**: Create this file in `~/thehive/thehive/config` with the following content:

    ```conf
    include "/etc/thehive/secret.conf"

    # DATABASE CONFIGURATION

>     db.janusgraph {
>       storage {
>           backend = cql
>           hostname = ["cassandra"]
>           cql {
>           cluster-name = thehive
>           keyspace = thehive
>           }
>       }
>       index.search {
>           backend = elasticsearch
>           hostname = ["elasticsearch"]
>           index-name = thehive
>       }
>     }

Reviewer: fix indentation
```suggestion
    db.janusgraph {
      storage {
        backend = cql
        hostname = ["cassandra"]
        cql {
          cluster-name = thehive
          keyspace = thehive
        }
      }
      index.search {
        backend = elasticsearch
        hostname = ["elasticsearch"]
        index-name = thehive
      }
    }
```

    # ATTACHMENTS STORAGE CONFIGURATION
    storage {
      ## Local filesystem
      provider: localfs
      localfs {
        location: /opt/thp/thehive/files
      }
    }

    # SERVICE CONFIGURATION
    stream.longPolling.refresh: 45 seconds
    play.http.context : "/"
    play.http.parser.maxDiskBuffer: 1GB
    play.http.parser.maxMemoryBuffer: 256kB
    ### Stream configuration
    stream {
      get {
        maxAttempts = 5
        minBackoff = 100 milliseconds
        maxBackoff = 500 milliseconds
        randomFactor = 0.2
      }
    }

    # CONNECTORS
    # Enable Cortex connector
    scalligraph.modules += org.thp.thehive.connector.cortex.CortexModule
    # Enable MISP connector
    scalligraph.modules += org.thp.thehive.connector.misp.MispModule
    ```

    &nbsp;

    **`logback.xml`**: Create this file in `~/thehive/thehive/config` with the following content:

    ```xml

>     <?xml version="1.0" encoding="UTF-8"?>
>     <configuration debug="false">
> 
>         <conversionRule conversionWord="coloredLevel"
>                         converterClass="play.api.libs.logback.ColoredLevel"/>
> 
>         <appender name="FILE" class="ch.qos.logback.core.rolling.RollingFileAppender">
>             <file>${application.home:-.}/logs/application.log</file>
>             <rollingPolicy class="ch.qos.logback.core.rolling.FixedWindowRollingPolicy">
>                 <fileNamePattern>${application.home:-.}/logs/application.%i.log.zip</fileNamePattern>
>                 <minIndex>1</minIndex>
>                 <maxIndex>10</maxIndex>
>             </rollingPolicy>
>             <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
>                 <maxFileSize>10MB</maxFileSize>
>             </triggeringPolicy>
>             <encoder>
>                 <pattern>%date [%level] from %logger in %thread %replace(\(%X{userId}@%X{organisation}\) ){'\(@\)
>                     ',''}[%X{kamonTraceId}] %message%n%xException
>                 </pattern>
>             </encoder>
>         </appender>
> 
>         <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
>             <encoder>
>                 <pattern>%date %coloredLevel %logger{15} %replace(\(%X{userId}@%X{organisation}\) ){'\(@\)
>                     ',''}[%X{kamonTraceId}] %message%n%xException{10}
>                 </pattern>
>             </encoder>
>         </appender>
> 
>         <appender name="ASYNCFILE" class="ch.qos.logback.classic.AsyncAppender">
>             <appender-ref ref="FILE"/>
>         </appender>
> 
>         <appender name="ASYNCSTDOUT" class="ch.qos.logback.classic.AsyncAppender">
>             <appender-ref ref="STDOUT"/>
>         </appender>
> 
>         <!-- do not set the following logger to TRACE -->
>         <logger name="org.thp.scalligraph.traversal" level="INFO"/>
>         <logger name="org.reflections.Reflections" level="ERROR"/>
>         <logger name="org.janusgraph.graphdb.database.management.ManagementLogger" level="OFF"/>
>         <logger name="org.janusgraph.graphdb.database.IndexSerializer" level="ERROR"/>
>         <logger name="org.apache.kafka" level="WARN"/>
>         <logger name="org.docx4j" level="WARN"/>
>         <logger name="org.docx4j.model.PropertyResolver" level="OFF" />
>         <!-- Log Elasticsearch requests -->
>         <!--<logger name="org.janusgraph.diskstorage.es.rest.RestElasticSearchClient" level="TRACE"/>-->
>         <logger name="org.thp.scalligraph.models" level="INFO"/>
>         <logger name="org.thp" level="INFO"/>
> 
>         <root level="INFO">
>             <appender-ref ref="ASYNCFILE"/>
>             <appender-ref ref="ASYNCSTDOUT"/>
>         </root>
> 
>     </configuration>

Reviewer: This logback.xml is the production one (not dev):
```suggestion
    <?xml version="1.0" encoding="UTF-8"?>
    <configuration debug="false">

        <conversionRule conversionWord="coloredLevel"
                        converterClass="play.api.libs.logback.ColoredLevel"/>

        <appender name="FILE" class="ch.qos.logback.core.rolling.RollingFileAppender">
            <file>/var/log/thehive/application.log</file>
            <rollingPolicy class="ch.qos.logback.core.rolling.FixedWindowRollingPolicy">
                <fileNamePattern>/var/log/thehive/application.%i.log.zip</fileNamePattern>
                <minIndex>1</minIndex>
                <maxIndex>10</maxIndex>
            </rollingPolicy>
            <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
                <maxFileSize>10MB</maxFileSize>
            </triggeringPolicy>
            <encoder>
                <pattern>%date [%level] from %logger in %thread %replace(\(%X{userId}@%X{organisation}\) ){'\(@\) ',''}[%X{kamonTraceId}] %message%n%xException</pattern>
            </encoder>
        </appender>

        <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
            <encoder>
                <pattern>%coloredLevel %logger{15} %replace(\(%X{userId}@%X{organisation}\) ){'\(@\) ',''}[%X{kamonTraceId}] %message%n%xException{10}
                </pattern>
            </encoder>
        </appender>

        <appender name="ASYNCFILE" class="ch.qos.logback.classic.AsyncAppender">
            <appender-ref ref="FILE"/>
        </appender>

        <appender name="ASYNCSTDOUT" class="ch.qos.logback.classic.AsyncAppender">
            <appender-ref ref="STDOUT"/>
        </appender>

        <!-- do not set the following logger to TRACE -->
        <logger name="org.thp.scalligraph.traversal" level="INFO"/>
        <logger name="org.reflections8.Reflections" level="ERROR" />
        <logger name="org.janusgraph.graphdb.database.management.ManagementLogger" level="OFF"/>
        <logger name="org.janusgraph.graphdb.database.IndexSerializer" level="ERROR"/>
        <logger name="org.apache.kafka" level="WARN" />
        <logger name="org.docx4j" level="WARN"/>
        <logger name="org.docx4j.model.PropertyResolver" level="OFF" />

        <logger name="org.thp" level="INFO"/>

        <root level="INFO">
            <appender-ref ref="ASYNCFILE"/>
            <appender-ref ref="ASYNCSTDOUT"/>
        </root>

    </configuration>
```

    ```

    &nbsp;

    **`secret.conf`**: Use the following commands to create the file:

>     ```bash
>     key=$(dd if=/dev/urandom bs=1024 count=1 | tr -dc 'a-zA-Z0-9' | fold -w 64 | head -n 1)
>     echo "play.http.secret.key=\"$key\"" | sudo tee -a ~/thehive/thehive/config/secret.conf
>     ```

Reviewer: Root permission is not needed here:
```suggestion
    ```bash
    key=$(dd if=/dev/urandom bs=1024 count=1 | tr -dc 'a-zA-Z0-9' | fold -w 64 | head -n 1)
    echo "play.http.secret.key=\"$key\"" > ~/thehive/thehive/config/secret.conf
    ```
```
